flocker/node/agents/functional/test_cinder_behaviour.py

```diff
@@ -112,3 +127,39 @@ def test_updated_metadata_is_listed(self):
         missing_items = expected_items - actual_items
 
         self.assertEqual(set(), missing_items)
+
+
+class BlockDeviceAPIDestroyTests(SynchronousTestCase):
+    """
+    Test for ``cinder.CinderBlockDeviceAPI.destroy_volume``
+    """
+    def setUp(self):
+        self.cinder_volumes = cinder_volume_manager()
+
+    def test_destroy_timesout(self):
+        """
+        If the cinder cannot delete the volume, we should timeout
+        after waiting some time
+        """
+        # Adding a fake get that will always return a volume, so we
+        # timeout waiting for the volume to be deleted
+        self.patch(self.cinder_volumes, "get", fakeGet)
+        # Using a fake no-op delete so it doens't actually delete anything
+        # (we don't need any actual volumes here, as we only need to verify
+        # the timeout)
+        self.patch(self.cinder_volumes, "delete", lambda *args, **kwargs: None)
+
+        api = CinderBlockDeviceAPI(
+            cinder_volume_manager=self.cinder_volumes,
+            nova_volume_manager=object(),
+            nova_server_manager=object(),
+            cluster_id=uuid4()
+            )
+        # Setting the timeut to 1, as the default is quite high, and we do not
+        # want to wait that much in a test
+        api.timeout = 1
```

Comment:
The tragedy of blocking APIs... A fake clock on which you controlled the passage of time would make this test run about 1 second faster.  It's hard to do that in this case, though, since you'd need to use an extra thread to control the clock (or run the delete, one or the other).

Separately, making the timeout a parameter to `__init__` would be better.  Setting attributes on other objects is always risky (maybe  you misspell the name, maybe the implementation changes so the attribute isn't used anymore, etc) and the failures are often hard to track down.